aptos-move/framework/aptos-framework/sources/configs/consensus_config.move

```diff
@@ -23,7 +25,11 @@ module aptos_framework::consensus_config {
         move_to(aptos_framework, ConsensusConfig { config });
     }
 
-    /// This can be called by on-chain governance to update on-chain consensus configs.
+    /// Deprecated by `set_for_next_epoch()`.
+    ///
+    /// WARNING: calling this while randomness is enabled will trigger a new epoch without randomness!
```

Comment:
Should we add checks in `set` and `set_for_next_epoch`, to make sure the conditions are met? like `set` should be called only when randomness is not enabled. If a governance proposal mistakenly uses `set`, this prevents entering a new epoch without randomness. On the other hand, using `set_for_next_epoch` without randomness enabled will never change epoch?

Reply:
`set_for_next_epoch(); reconfigure();` combination is supposed to be the only API.
. but we have some tests relying on `set()`, as well as some genesis usage, so currently it's hard to even update them.

But yes, eventually it should be what you described.